packages/flutter/lib/src/material/slider.dart

```diff
@@ -0,0 +1,146 @@
+
+  final double value;
+  final ValueChanged<double> onChanged;
+
+  Widget build(BuildContext context) {
+    return new _SliderWrapper(
+      value: value,
+      primaryColor: Theme.of(context).accentColor,
+      onChanged: onChanged
+    );
+  }
+}
+
+class _SliderWrapper extends LeafRenderObjectWidget {
+  _SliderWrapper({ Key key, this.value, this.primaryColor, this.onChanged })
+      : super(key: key);
+
+  final double value;
+  final Color primaryColor;
+  final ValueChanged<double> onChanged;
+
+  _RenderSlider createRenderObject() => new _RenderSlider(
+    value: value,
+    primaryColor: primaryColor,
+    onChanged: onChanged
+  );
+
+  void updateRenderObject(_RenderSlider renderObject, _SliderWrapper oldWidget) {
+    renderObject.value = value;
+    renderObject.primaryColor = primaryColor;
+    renderObject.onChanged = onChanged;
+  }
+}
+
+final Color _kInactiveTrackColor = Colors.grey[400];
+final Color _kActiveTrackColor = Colors.grey[500];
+
+class _RenderSlider extends RenderConstrainedBox {
+  _RenderSlider({
+    double value,
+    Color primaryColor,
+    this.onChanged
+  }) : _value = value,
+       _primaryColor = primaryColor,
+        super(additionalConstraints: const BoxConstraints.tightFor(width: 144.0, height: 2 * _kReactionRadius)) {
+    _drag = new HorizontalDragGestureRecognizer(router: FlutterBinding.instance.pointerRouter)
+      ..onStart = _handleDragStart
+      ..onUpdate = _handleDragUpdate
+      ..onEnd = _handleDragEnd;
+  }
+
+  double get value => _value;
+  double _value;
+  void set value(double newValue) {
+    if (newValue == _value)
+      return;
+    _value = newValue;
+    markNeedsPaint();
+  }
+
+  Color get primaryColor => _primaryColor;
+  Color _primaryColor;
+  void set primaryColor(Color value) {
+    if (value == _primaryColor)
+      return;
+    _primaryColor = value;
+    markNeedsPaint();
+  }
+
+  ValueChanged<double> onChanged;
+
+  HorizontalDragGestureRecognizer _drag;
+  bool _active = false;
+  double _currentDragValue = 0.0;
+
+  void _handleDragStart(Point globalPosition) {
+    _active = true;
+    markNeedsPaint();
+    if (onChanged != null) {
+      _currentDragValue = globalToLocal(globalPosition).x / size.width;
+      onChanged(_currentDragValue.clamp(0.0, 1.0));
+    }
+  }
+
+  void _handleDragUpdate(double delta) {
+    if (onChanged != null) {
+      _currentDragValue += delta / size.width;
+      onChanged(_currentDragValue.clamp(0.0, 1.0));
+    }
+  }
+
+  void _handleDragEnd(Offset velocity) {
+    _active = false;
+    _currentDragValue = 0.0;
+    markNeedsPaint();
+  }
+
+  bool hitTestSelf(Point position) => true;
+
+  void handleEvent(InputEvent event, BoxHitTestEntry entry) {
+    if (event.type == 'pointerdown')
+      _drag.addPointer(event);
+  }
+
+  void paint(PaintingContext context, Offset offset) {
+    final Canvas canvas = context.canvas;
+
+    double trackCenter = offset.dy + size.height / 2.0;
+    double trackLeft = offset.dx;
```

Comment:
you need some padding on this guy or you won't be able to hit the left side of the knob when it's on the left and the right side of the knob when it's on the right.

I guess you need to add 2*kReactionRadius to the width, then paint at 1*kReactionRadius in from the offset.dx, or some such.